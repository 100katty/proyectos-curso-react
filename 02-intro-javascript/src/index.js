console.log('hola mundo');


const nombre = 'catalina'
const apellido = 'hernandes';

const nombreCompleto = nombre + '' + apellido;

console.log(nombreCompleto);