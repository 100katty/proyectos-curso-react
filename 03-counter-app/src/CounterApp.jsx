import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);
    
    const handleAdd=()=> {
        setCounter(counter + 1);
       // setCounter((c)=> c + 20)
    }
    const handleRest=()=> {
        setCounter(counter - 1);
       // setCounter((c)=> c + 20)
    }
    const handleReset=()=> {
        setCounter(value);
       // setCounter((c)=> c + 20)
    }
    

  return ( 
      <>
          <h1>CounterApp</h1>
          <h2> {counter} </h2>
          <button onClick={handleAdd} >
              +1
          </button>
          <button onClick={handleRest} >
              -1
          </button>
          <button onClick={handleReset} >
              Reset
          </button>
      </>
  )
}

CounterApp.Prototype = {
    value: PropTypes.number.isRequired
   
}