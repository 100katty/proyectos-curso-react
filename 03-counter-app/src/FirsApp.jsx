import React from 'react'

export const FirsApp =()=> {
    return (
        <>
            <h1>FirsApp</h1>
            <p>Soy un subititulo</p>
        </>
      
  )
}
