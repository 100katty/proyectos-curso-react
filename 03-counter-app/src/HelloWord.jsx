import React from 'react'

export const  HelloWord=() =>{
  return (
    <h1>HelloWord App</h1>
  )
}
