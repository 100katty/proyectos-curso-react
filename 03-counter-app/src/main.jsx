import React from 'react';
import ReactDOM from 'react-dom/client';
//import {HelloWordApp} from './HelloWord';
//import { FirsApp } from './FirsApp';
import { CounterApp } from './CounterApp';
import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
         <CounterApp value={20} /> 
        
        {/* <FirsApp title="hola, soy mundo" />*/}
    </React.StrictMode>
);