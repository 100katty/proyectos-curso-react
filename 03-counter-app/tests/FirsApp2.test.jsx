import { render ,screen} from "@testing-library/react";
import { FirsApp } from "../src/FirsApp";

describe('Pruebas en <FirstApp />', () => {
    const title = 'Hola spy mundo';
    const subtitle = 'katty';
    

    test('debe de hacer match con el snapshot', () => {
    const {container} = render( <FirsApp title={title} subtitle={subtitle} />)
    expect(container).toMatchSnapshot();
        
    });

    test('debe de de mostrar el mensaje hola mundo', () => {
        
        render( <FirsApp title={title}  />)
        expect(screen.getByText(title)).toBeTruthy();
        //screen.debug();
        
    });
    
     test('debe de de mostrar el title en un h1', () => {
        
        
        render( <FirsApp title={title}  />)
        expect(screen.getByRole('heading',{level:1}).innerHTML ).toContain(title)


        
    });


     test('debe de de mostrar el subtitle enviado por prps', () => {
        
        
         render(
              <FirsApp title={title}
                 subtitle={subtitle} />
         )
         expect(screen.getAllByText(subtitle).length).toBe(2);



         

        
    });
});
