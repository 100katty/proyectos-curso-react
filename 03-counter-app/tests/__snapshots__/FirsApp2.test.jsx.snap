// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola spy mundo
     
  </h1>
  <p>
    katty
     
  </p>
  <p>
    katty
     
  </p>
</div>
`;
